In [1]:
#LATIHAN 1
import numpy as np

harga = np.array([5000, 7000, 3000, 12000, 4500])
print('Rata-rata harga:', harga.mean())
print('Harga tertinggi:', harga.max())
print('Harga setelah diskon 10%', harga * 0.9)
print('Harga setelah diskon 10%', [5000, 7000, 3000] * 0.9)

Rata-rata harga: 6300.0
Harga tertinggi: 12000
Harga setelah diskon 10% [ 4500.  6300.  2700. 10800.  4050.]


TypeError: can't multiply sequence by non-int of type 'float'

Tugas Analisis 1: Bandingkan hasil harga * 0.9 di atas dengan jika harga berupa list Python
biasa (bukan array). Apa yang terjadi bila kamu mencoba [5000,7000,3000] * 0.9? Jelaskan
mengapa berbeda.

Akan ada error, karena list python dibaca bukan sebagai array integer. List tersebut dibaca sebagai non-int dan tidak bisa masuk dalam operasi matematika dengan tipe float (0.9)

In [2]:
#LATIHAN 2
import pandas as pd

data_kantin = {
    'menu': ['Nasi Goreng', 'Es Teh', 'Mie Ayam', 'Es Teh', None],
    'harga': [12000, 4000, 10000, 4000, 8000],
    'terjual': [23, 40, None, 35, 18]
}
df = pd.DataFrame(data_kantin)
print(df)

          menu  harga  terjual
0  Nasi Goreng  12000     23.0
1       Es Teh   4000     40.0
2     Mie Ayam  10000      NaN
3       Es Teh   4000     35.0
4          NaN   8000     18.0


Tugas Analisis 2: Amati tabel yang dihasilkan. Kolom mana yang terlihat memiliki data kosong
(None)? Menurutmu apa risikonya jika data ini langsung dianalisis tanpa dibersihkan?

Terdapat data kosong di kolom 'terjual' pada index 2. ini akan menyebabkan error dalam pemrosesan dan penghitungan data karena data kosong tetap dibiarkan dan tidak ada operator untuk mengisi atau menghapus data kosong

In [3]:
#LATIHAN 3
import pandas as pd

df = pd.read_csv(r'C:\Users\Reza Al Fath\Documents\EDA_python\data_kantin.csv')

print(df.head()) # 5 baris pertama
print(df.info()) # tipe data & jumlah non-null tiap kolom
print(df.describe()) # statistik ringkas kolom numerik
print(df.shape) # jumlah (baris, kolom)

      tanggal         menu kategori  harga  terjual
0  2026-08-20   Teh Hangat  Minuman   3000     11.0
1  2026-08-31  Nasi Goreng  Makanan  12000     29.0
2  2026-08-21  Nasi Goreng  Makanan  12000     28.0
3  2026-08-28   Teh Hangat  Minuman   3000     12.0
4  2026-08-21     Es Jeruk  Minuman   5000     19.0
<class 'pandas.DataFrame'>
RangeIndex: 110 entries, 0 to 109
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   tanggal   110 non-null    str    
 1   menu      107 non-null    str    
 2   kategori  110 non-null    str    
 3   harga     110 non-null    int64  
 4   terjual   104 non-null    float64
dtypes: float64(1), int64(1), str(3)
memory usage: 4.4 KB
None
              harga     terjual
count    110.000000  104.000000
mean    6336.363636   22.923077
std     3520.001232   10.831178
min     2000.000000    8.000000
25%     3000.000000   13.000000
50%     5000.000000   20.000000
75%    10000.000000   33.000000
max

Tugas Analisis 3: Dari hasil df.info(), kolom mana yang jumlah non-null-nya lebih sedikit dari
jumlah baris total? Apa artinya?

Kolom menu dan terjual. Karena bisa dilihat dari RangeIndex (total kolom), kolom menu hanya memiliki 107 non-null yang menandakan ada 3 kolom dengan null value. Sama juga dengan kolom terjual yang memiliki 6 kolom null value.

In [4]:
#LATIHAN 4
print(df.isnull().sum()) # jumlah data kosong tiap kolom
df['terjual'] = df['terjual'].fillna(0) # isi kekosongan dengan 0
df = df.dropna(subset=['menu'])

tanggal     0
menu        3
kategori    0
harga       0
terjual     6
dtype: int64


Tugas Analisis 4: Mengapa pada contoh di atas kolom terjual diisi (fillna) sedangkan baris
dengan menu kosong justru dihapus (dropna)? Diskusikan alasan logisnya.

Karena untuk kolom terjual, kita bisa menganggap 0 mewakili jika produk tersebut tidak laku/terdapat transaksi, tapi kita juga ingin memastikan agar kita masih bisa melakukan penghitungan hasil akhir penjualan. Itu sebabnya kita mengisi nilai null dengan 0.

Untuk menu kosong, kita mengambil keputusan untuk menghapusnya karena itu sama saja dengan kita tidak menjual menu itu. Jika kita membiarkan null ndi kolom menu, itu akan membuat nilai kolom lain terasa dirasa anonim (aneh jika kita mengetahui nilai harga atau terjual tanpa mengetahui apa nama menunnya)

In [5]:
#LATIHAN 5
print(df.duplicated().sum()) # jumlah baris duplikat
df = df.drop_duplicates()
df['harga'] = df['harga'].astype(int) # memastikan tipe data harga adalah integer
print(df.dtypes)

5
tanggal         str
menu            str
kategori        str
harga         int64
terjual     float64
dtype: object


In [6]:
print("Jumlah baris sekarang:", len(df))

Jumlah baris sekarang: 102


Tugas Analisis 5: Setelah drop_duplicates(), berapa jumlah baris dataset sekarang dibanding
sebelumnya? Mengapa penting memastikan tipe data (dtypes) sudah benar sebelum data
dianalisis lebih lanjut?

Jumlah dataset sekarang adalah 102 dengan jumlah sebelumnya adalah 107 baris
Jika angka bertipe string (teks), penjumlahan akan menggabungkan teks


In [ ]:
#LATIHAN 6
laris = df[df['terjual'] > 20] # filtering
urut = df.sort_values(by='terjual', ascending=False) # sorting
df['total_pendapatan'] = df['harga'] * df['terjual'] # kolom turunan
ringkasan = df.groupby('menu')['total_pendapatan'].sum() # agregasi
print(ringkasan)

menu
Es Jeruk       1460000.0
Es Teh         2144000.0
Kerupuk         972000.0
Mie Ayam       2410000.0
Nasi Goreng    3924000.0
Roti Bakar     1288000.0
Teh Hangat      459000.0
Name: total_pendapatan, dtype: float64


Tugas Analisis 6: Dari hasil groupby di atas, menu apa yang menghasilkan total_pendapatan
tertinggi? Bagaimana informasi ini bisa membantu pengambilan keputusan di kantin sekolah?

Menu nasi goreng menghasilkan total_pendapatan tertinggi dengan total 3.924.000
Informasi ini berguna jika semisal kita ingin mengetahui sumber pendapatan kantin terbesar. Memberikan perhatian ekstra pada kecepatan dan kualitas penyajian Nasi Goreng karena merupakan cash cow (penyumbang pemasukan terbesar) bagi kantin.